src/vm/comsynchronizable.cpp

```diff
@@ -1537,15 +1537,56 @@ void QCALLTYPE ThreadNative::nativeInitCultureAccessors()
     END_QCALL;
 }
 
+typedef HRESULT(WINAPI *pfnSetThreadDescription)(HANDLE hThread, PCWSTR lpThreadDescription);
+extern pfnSetThreadDescription g_pfnSetThreadDescription;
+BOOL g_InitializeSetThreadDescriptionWasRun = FALSE;
+
+
+HRESULT WINAPI InitializeSetThreadDescription(HANDLE hThread, PCWSTR lpThreadDescription)
+{
+    pfnSetThreadDescription func = NULL;
+
+#ifndef FEATURE_PAL
+    if (!g_InitializeSetThreadDescriptionWasRun)
+    {
+        HMODULE hKernel32 = WszLoadLibrary(W("kernel32.dll"));
+        if (hKernel32 != NULL)
+        {
+            func = (pfnSetThreadDescription)GetProcAddress(hKernel32, "SetThreadDescription");
+        }
+        g_InitializeSetThreadDescriptionWasRun = TRUE;
```

Comment:
Still does not look right - when we get here second time on Windows Creators update, `g_InitializeSetThreadDescriptionWasRun` will be true and so func will remain null.